### PR TITLE
raise error on dispatching a non-lattice

### PR DESCRIPTION
**Pull request by madhur-tandon**
Fixes https://github.com/AgnostiqHQ/covalent/issues/793

- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by madhur-tandon** [comment]
CC: @santoshkumarradha @AlejandroEsquivel

File: covalent/_dispatcher_plugins/local.py [hunk 1]
```diff
@@ -122,6 +122,11 @@ def wrapper(*args, **kwargs) -> str:
             # To access the disable_run passed to the dispatch function
             nonlocal disable_run
 
+            if not isinstance(orig_lattice, Lattice):
+                message = f"Dispatcher expected a Lattice, received {type(orig_lattice)} instead."
+                app_log.error(message)
+                raise ValueError(message)
```
Comment (kessler-frost) [inline covalent/_dispatcher_plugins/local.py:128]: I think a `TypeError` would be better here.

**Review by kessler-frost — CHANGES_REQUESTED** [review]
Just a minor comment, once that's resolved this is gtg 🚀 !

File: covalent/_dispatcher_plugins/local.py (hunk 1 above)
Comment (madhur-tandon) [reply covalent/_dispatcher_plugins/local.py]: Aah, sorry, my bad, `TypeError` it should be.

**Comment by madhur-tandon** [comment]
@kessler-frost seems like `license/cla` is stuck?

**Comment by kessler-frost** [comment]
Yeah, I wouldn't worry about it, it should get done soon. Automerge is enabled so once these checks pass this will get merged automatically.

**Review by kessler-frost — DISMISSED** [review]

**Comment by madhur-tandon** [comment]
> Yeah, I wouldn't worry about it, it should get done soon. 

I think you might have to re-trigger this because I don't remember this taking so long... 😅

**Comment by madhur-tandon** [comment]
@kessler-frost I think you triggered `license` while it should have been `license/cla` i.e. the contributor's license agreement.

**Comment by kessler-frost** [comment]
Yeah I just realized that, my bad. Can you try making another commit on this one? That usually fixes it.

**Comment by kessler-frost** [comment]
Yep, that fixed it, thanks! (For some reason I couldn't find that action 🤔 )

**Review by kessler-frost — APPROVED** [review]

**Pull request merged**